invisible_cities/reco/icaro_components.py

```diff
@@ -0,0 +1,205 @@
+
+def select_nS_mask_and_check(dst          : pd.DataFrame                       ,
+                             column       : type_of_signal                     ,
+                             input_mask   : Optional[np.ndarray]  = None       ,
+                             eff_interval : Tuple[float, float] = [0,1]        ,
+                             strictness   : Strictness = Strictness.raise_error
+                             )->np.ndarray:
+    """
+    Selects nS1(or nS2) == 1 for a given kr dst and
+    returns the mask. It also computes selection efficiency,
+    checking if the value is within a given interval, and
+    saves histogram parameters.
+    Parameters
+    ----------
+    dst: pd.Dataframe
+        Krypton dst dataframe.
+    column: type_of_signal
+        The function can be appplied over nS1 or nS2.
+    input_mask: np.array (Optional)
+        Selection mask of the previous cut. If this is the first selection
+        /no previous maks is input, input_mask is set to be an all True array.
+    interval: length-2 tuple
+        If the selection efficiency is out of this interval
+        the map production will abort/just warn, depending on "strictness".
+    sstrictness: Strictness
+        If 'warning', function returns a False if the criteria
+        is not matched. If 'stop_proccess' it raises an exception.
+    Returns
+    ----------
+        A mask corresponding to the selected events.
+    """
+    input_mask = input_mask if input_mask is not None else [True] * len(dst)
+    mask             = np.zeros_like(input_mask)
+    mask[input_mask] = dst.loc[input_mask, column.value] == 1
+
+    nevts_after      = dst[mask]      .event.nunique()
+    nevts_before     = dst[input_mask].event.nunique()
+    eff              = nevts_after / nevts_before
+    all_in_range(data         = np.array(eff)  ,
+                 minval       = eff_interval[0],
+                 maxval       = eff_interval[1],
+                 display_name = column.value   ,
+                 strictness   = strictness     ,
+                 right_closed = True)
+
+    return mask
+
+
+def select_band_and_check(dst         : pd.DataFrame                                     ,
+                          boot_map    : ASectorMap                                       ,
+                          norm_strat  : NormStrategy                   = NormStrategy.max,
+                          input_mask  : np.ndarray                     = None            ,
+                          range_DT    : Tuple[np.ndarray, np.ndarray]  = (10, 1300)      ,
+                          range_E     : Tuple[np.ndarray, np.ndarray]  = (10.0e+3,14e+3) ,
+                          nsigma_sel  : float                          = 3.5             ,
+                          eff_interval: Tuple[float, float]            = [0,1]           ,
+                          strictness  : Strictness = Strictness.raise_error
+                          )->np.array:
```

Comment:
Same comment about optionality. In this case, perhaps `nsigma_sel` can also be optional